pkg/controller/petset/iterator.go

```diff
@@ -0,0 +1,168 @@
+package petset
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/apps"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+// petGenerator generates pets.
+type petGenerator struct {
+	// ps is the petset for which this generator generates pets
+	ps *apps.PetSet
+	// idMappers are a list of identityMapper interfaces capable of giving
+	// the generates pets an identity
+	idMappers []identityMapper
+}
+
+// newPet generates a new pet using the id string as a unique qualifier
+func (pg *petGenerator) newPet(id string) (*pcb, error) {
+	petPod, err := controller.GetPodFromTemplate(&pg.ps.Spec.Template, pg.ps)
+	if err != nil {
+		return nil, err
+	}
+	for _, im := range pg.idMappers {
+		im.AddIdentity(id, petPod)
```

Comment:
"id" looks really confusing here. If renamed to "suffix", "AddSuffix", then it would be much clearer